431: Node upgrade v20

rpeterman-gp opened the pull request:
Closes #431 and #433 

- Upgrade tool-version to use Node v20
- Fix various rendering errors related to state hydration and HTML content rendering
- Fix bug where audio added via a sidebar list would result in player and playlist descriptions not linking to content
- Fix story fetch to include date prop
- Fix issue with footer menu links using incorrect route for some links
- Add caniuse update to dev:start script

## To Review

- [ ] Use the Preview link located in the Now comment below.

> ...or...

- [x] Checkout Branch.
- [x] Run `asdf install`.
- [x] Run `yarn`.
- [x] Run `yarn dev`.
- [x] Go to [localhost:3000](https://localhost:3000).

> ...then...

- [x] Ensure app loads and operates as expected.
- [x] Should no longer see unmounted component render warnings.
- [x] Should no longer see missing key prop warnings when rendering HTML content.
- [x] Add audio from sidebar list.
- [x] Ensure clicking on track info in player and playlist takes you to the audio's content page.
- [x] Click the team page link in footer menu.
- [ ] Ensure you console show compilation of the `/programs/[slug]/team` route and not `/[...alias]` route.

rpeterman-gp committed:
config: node v20 update

rpeterman-gp committed:
config: add caniuse update to dev:start

rpeterman-gp committed:
fix(app): state hydration render errors

rpeterman-gp committed:
fix(appfooter): menu use contentlink component

rpeterman-gp committed:
fix(htmlcontent): missing key prop error

- fix rendering block nodes in p tags
- fix to flatten deeply nested spans

rpeterman-gp committed:
fix(story): add date to fetched props

- change updated date to subtitle2 variant

rpeterman-gp committed:
fix(player): audio added via sidebar missing link

- improve centering of progress handle time element
- playlist images should not look squished

rpeterman-gp committed:
fix(ctaregion): content padding w/o card actions

rpeterman-gp committed:
fix(fonts): remove fallbacks

brandonhundt reviewed (commented):
@rpeterman-gp Noting the footer menu is not rendering for me. It appears to be set up in the CMS.

![Screenshot 2024-02-12 at 10 19 36 AM](https://github.com/PRX/theworld.org/assets/369739/293e3536-970a-423a-a7c6-fa87f8d66848)

rpeterman-gp committed:
docs(readme): update for wordpress development

- remove `./config` directory (not used anymore)
- update any files still using `@config` imports to use `.env` variables
- cleanup unused file path aliases
- update `.env.example` to include currently use variables w/ notes

brandonhundt reviewed (approved):
Thanks @rpeterman-gp All good now!
